recipe/meta.yaml

```diff
@@ -29,33 +29,42 @@ outputs:
       run_exports:
         - {{ pin_subpackage('openmpi', min_pin='x.x.x', max_pin='x') }}
       ignore_run_exports:
-        - ucx  # [enable_cuda]
+        - ucx  # [enable_ucx]
       script_env:
-        - CUDA_HOME     # [enable_cuda]
+        - CUDA_HOME  # [enable_cuda]
     requirements:
       build:
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
         - {{ compiler('fortran') }}
-        - autoconf  # [unix]
-        - automake  # [unix]
+        #- autoconf  # [unix]
+        #- automake  # [unix]
         - libtool   # [unix]
         - make  # [unix]
-        - perl 5.26.2
       host:
+        #- openpmix
+        #- prrte
+        - libhwloc
+        - libevent
         - zlib
-        - ucx                 # [enable_cuda]
-        - ucx-proc =*=gpu     # [enable_cuda]
+        - ucx  # [enable_ucx]
+        - cuda-version {{ cuda_version }}  # [enable_cuda]
       run:
-        - zlib
         - mpi 1.0 openmpi
+        #- openpmix
+        #- prrte
+        - libhwloc
+        - libevent
+        - zlib
       run_constrained:
-        - cudatoolkit  >= {{ cudatoolkit }}         # [enable_cuda]
```

Comment:
It might be worth keep `cudatoolkit` in `run_constrained` for good measure. Users may still try to install it (directly or indirectly) when using CUDA 11

FWIW here is how [`numba` constrains `cuda-version` & `cudatoolkit`]( https://github.com/conda-forge/numba-feedstock/blob/172c489224b795438fdbf289d04cbc188ddb480a/recipe/meta.yaml#L57-L59 )

Reply:
@jakirkham Why isn't the lock on `cuda-version` enough to constrain `cudatoolkit` ? Do you mean that there are old versions of `cudatoolkit` that ignore the `cuda-version` constraint?